core/src/com/mygdx/game/models/HexBoardClass.java

```diff
@@ -0,0 +1,113 @@
+package com.mygdx.game.models;
+
+import java.util.*;
+
+/*
+ * The hex board of Type T that we will use as the map of the game. T will be a terrain
+ */
+public class HexBoardClass<T> implements HexBoard {
+    private Object[][] board;
+    int xLength;
+    int yLength;
+
+    public HexBoardClass(Object o, int xLength, int yLength) {
+        board = new Object[xLength][yLength];
+        this.xLength = xLength;
+        this.yLength = yLength;
+
+        for (int i = 0; i < xLength; i++) {
```

Comment:
Here your innitializing every Hex as a string. This is unnessary.

Reply:
The set hex method should be used to initaliaze the values on the board.

Reply:
Or the set board method that you made. In the future we will need a way to initialize a board given a configuration file that represents a mission or map in the game.